src/DataWrangling/Copernicus/Copernicus.jl

```diff
@@ -60,12 +68,20 @@ copernicusmarine_dataset_id(::CMEMSHourlyAnalysis) = "cmems_mod_glo_phy_anfc_0.0
 CopernicusMetadata{D} = Metadata{<:CopernicusDataset, D}
 CopernicusMetadatum = Metadatum{<:CopernicusDataset}
 
-Base.size(::CopernicusMetadatum) = (4320, 2040, 50, 1)
+Base.size(::GLORYSStatic) = (4320, 2040, 50, 1)
+Base.size(::GLORYSDaily) = (4320, 2040, 50, 1)
+Base.size(::GLORYSMonthly) = (4320, 2040, 50, 1)
+Base.size(::GLORYSBGCDaily) = (1440, 680, 75, 1)
+Base.size(::GLORYSBGCMonthly) = (1440, 680, 75, 1)
 reversed_vertical_axis(::CopernicusDataset) = true
 
-available_variables(::CopernicusDataset) = copernicus_dataset_variable_names
+available_variables(::GLORYSStatic) = copernicus_physics_dataset_variable_names
+available_variables(::GLORYSDaily) = copernicus_physics_dataset_variable_names
+available_variables(::GLORYSMonthly) = copernicus_physics_dataset_variable_names
+available_variables(::GLORYSBGCDaily) = copernicus_bgc_daily_dataset_variable_names
+available_variables(::GLORYSBGCMonthly) = copernicus_bgc_monthly_dataset_variable_names
```

Comment:
how could we have `available_variables` give out different list of variables based on the dataset type if all variables are in one dictionary?

Reply:
@glwagner does this look ok now?

Reply:
yes, thank you! I can eliminate the repeated code by creating a more organized type hierarchy. do you mind if I commit to your branch?

Reply:
Go for it!